src/resources/formats/html/ojs/ojs-bundle.js

```diff
@@ -708,8 +731,18 @@ export function createRuntime() {
   });
   quartoOjsGlobal.obsInABox = obsInABox;
   if (isShiny) {
+    // When isShiny, OJSInABox is constructed with allowPendingGlobals:true.
+    // Our guess is that most shiny-to-ojs exports will have been defined
+    // by the time the server function finishes executing (i.e. session init
+    // completion); so we call `killPendingGlobals()` to show errors for
+    // variables that are still undefined.
```

Comment:
This is the point where we would handle things differently if we decide to use the `ojs-imports` YAML entry to explicitly encode the expected Shiny outputs, right?

Reply:
Yes, although I don't think there's any reason why we can't allow both simultaneously. (I haven't tested that)